lib/presentation/widget/fav_button.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+
+import '../../core/util/constants/ui_constants.dart';
+import '../../core/util/notifications/notification_service.dart';
+import '../bloc/movies_bloc.dart';
+
+class FavButton extends StatefulWidget {
+  const FavButton({
+    super.key,
+    required this.title,
+    required this.id,
+  });
+
+  final String title;
+  final int id;
+
+  @override
+  State<FavButton> createState() => _FavButtonState();
+}
+
+class _FavButtonState extends State<FavButton> {
+  static const String removeActionText = 'Remove From Favorites';
+  static const String addActionText = 'Add to Favorites';
+  static const String notificationTitle = 'TheMovieDB';
+  static const String removedNotificationBody = ' Removed from favorites';
+  static const String addedNotificationBody = 'Added to favorites';
+
+  bool isFav() =>
```

Comment:
Remember to be more descriptive on method/variables names, you can rename this, like: `isFavouriteMovie`

Reply:
done